### PR TITLE
Add manual_execution param to test runs controller

### PR DESCRIPTION
**Pull request by wwwwhiterabbit**

### CODE REVIEW

File: app/controllers/test_runs_controller.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def create
 
     respond_to do |format|
       if @test_run.save
-        TestRunJob.perform_later(@test_run.id)
+        TestRunJob.perform_later(@test_run.id) unless manual_execution?
```
Comment (wwwwhiterabbit) [inline app/controllers/test_runs_controller.rb:35]: need to update the `performed` attribute

File: app/controllers/test_model_version_runs_controller.rb [hunk 2]
```diff
@@ -19,4 +33,16 @@ def set_test_model_version_run
...
+    ApplicationRecord.transaction do
+      @test_model_version_run.lock!
+      break false if @test_model_version_run.performed?
+
+      jobs = Array.new(@test_run.calls) { TestModelVersionRunJob.new(@test_model_version_run.id) }
+      SolidQueue::Job.enqueue_all(jobs)
+
+      @test_model_version_run.update(performed: true)
+    end
+  end
```
Comment (wwwwhiterabbit) [inline app/controllers/test_model_version_runs_controller.rb:47]: move this method to service class?

Comment (Vadser) [reply app/controllers/test_model_version_runs_controller.rb:47]: I think yes, move it

File: app/services/perform_test_model_version_run_jobs.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class PerformTestModelVersionRunJobs
+  attr_reader :test_run_id, :test_model_version_run_id
+
+  def initialize(test_run_id, test_model_version_run_id)
```
Comment (Vadser) [inline app/services/perform_test_model_version_run_jobs.rb:6]: I think we can use model for test_run in initializer

File: app/controllers/test_model_version_runs_controller.rb [hunk 4]
```diff
@@ -34,15 +34,7 @@ def set_test_run
...
-      @test_model_version_run.lock!
-      break false if @test_model_version_run.performed?
-
-      jobs = Array.new(@test_run.calls) { TestModelVersionRunJob.new(@test_model_version_run.id) }
-      SolidQueue::Job.enqueue_all(jobs)
-
-      @test_model_version_run.update(performed: true)
-    end
+  def perform_test_model_version_run_jobs
+    PerformTestModelVersionRunJobs.new(@test_run.id, @test_model_version_run.id).call
```
Comment (wwwwhiterabbit) [inline app/controllers/test_model_version_runs_controller.rb:38]: I don't like these extra queries being executed in the before_action hooks for all endpoints. We only need to check if the current user has access to the test_run with the given ids.

File: app/controllers/test_model_version_runs_controller.rb (hunk 2 above)
Comment (wwwwhiterabbit) [reply app/controllers/test_model_version_runs_controller.rb:39]: done

File: app/services/perform_test_model_version_run_jobs.rb (hunk 3 above)
Comment (wwwwhiterabbit) [reply app/services/perform_test_model_version_run_jobs.rb]: pls have a look at https://github.com/aifoundry-org/llamagator/pull/55/commits/0fb0092cccfa455a72045220e307a17de3a4bf55

File: app/controllers/test_model_version_runs_controller.rb (hunk 4 above)
Comment (wwwwhiterabbit) [reply app/controllers/test_model_version_runs_controller.rb]: skipped `set_test_model_version_run` hook for the `perform` action

**Review by anatoliliotych — CHANGES_REQUESTED** [review]

File: db/migrate/20240820104429_add_performed_to_test_model_version_run.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddPerformedToTestModelVersionRun < ActiveRecord::Migration[7.1]
+  def change
+    add_column :test_model_version_runs, :performed, :boolean, null: false, default: false
```
[the remark this answers is not in the document]
Comment (wwwwhiterabbit) [reply db/migrate/20240820104429_add_performed_to_test_model_version_run.rb]: pls check this commit https://github.com/aifoundry-org/llamagator/pull/55/commits/ad195cd522f03fe3a2bdcbbe9b3388eac939928d

**Review by anatoliliotych — APPROVED** [review]

**Pull request merged**